app/controllers/search_controller.rb

```diff
@@ -210,6 +248,20 @@ def search_samples
     @search_count = @sample_search_count
   end
 
+  def search_repository
+    @repository = Repository.find_by_id(params[:repository])
+    render_403 unless can_view_repository(@repository)
```

Comment:
Because here we get id of repository directly from the user, in other cases we get only query